utils/cagg_refresh_producer_consumer/producer.sql

```diff
@@ -0,0 +1,115 @@
+CREATE SCHEMA IF NOT EXISTS _timescaledb_additional;
+
+CREATE TABLE IF NOT EXISTS _timescaledb_additional.incremental_continuous_aggregate_refreshes (
+    continuous_aggregate regclass not null,
+    window_start timestamptz not null,
+    window_end timestamptz not null CHECK (window_end > window_start),
+    scheduled timestamptz not null default pg_catalog.clock_timestamp(),
+    worker_pid integer
+);
```

Comment:
Shouldn't we need a column to know when it was processed??? Or it will be deleted?

Reply:
> Shouldn't we need a column to know when it was processed

Let's do that, then we can create some reports afterwards.
We do need to think about retention that way though, especiallly if the number of tasks is large.

Reply:
```
    started timestamptz,
    finished timestamptz,
```

Adding these, and changing unique index:

```
) WHERE worker_pid IS NULL AND finished IS NULL;
```

Reply:
We can turn it into a hypertable then to make our life easier to apply retention policies

Reply:
Alas, then my trick with the unique index won't work!